milli/src/update/delete_documents.rs

```diff
@@ -93,6 +96,50 @@ impl<'t, 'u, 'i> DeleteDocuments<'t, 'u, 'i> {
         }
 
         let fields_ids_map = self.index.fields_ids_map(self.wtxn)?;
+        let mut field_distribution = self.index.field_distribution(self.wtxn)?;
+
+        // we update the field distribution
+        for docid in self.to_delete_docids.iter() {
+            let key = BEU32::new(docid);
+            let document =
+                self.index.documents.get(self.wtxn, &key)?.ok_or(
+                    InternalError::DatabaseMissingEntry { db_name: "documents", key: None },
+                )?;
+            for (fid, _value) in document.iter() {
+                let field_name =
+                    fields_ids_map.name(fid).ok_or(FieldIdMapMissingEntry::FieldId {
+                        field_id: fid,
+                        process: "delete documents",
+                    })?;
+                if let Entry::Occupied(mut entry) = field_distribution.entry(field_name.to_string())
+                {
+                    match entry.get().checked_sub(1) {
+                        Some(0) | None => entry.remove(),
+                        Some(count) => entry.insert(count),
+                    };
+                }
+            }
+        }
```

Comment:
I would prefer that we don't add this part of the code here, I would prefer that we store the docids corresponding to each field in the field distribution instead. This way when we retrieve the distribution of the fields we can simply remove the soft-deleted once. Could you change that and add a commit for the previously described behavior, please?

Reply:
Hum... in fact could you please open an issue about doing that later?